plugins/hydra_tsp_launcher/hydra_plugins/hydra_tsp_launcher/tsp_launcher.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from pathlib import Path
+from typing import Optional, Sequence, Any, Dict, List
+
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.config_search_path import ConfigSearchPath
+from hydra.core.hydra_config import HydraConfig
+from hydra.core.singleton import Singleton
+from hydra.core.utils import (
+    JobReturn,
+    configure_log,
+    filter_overrides,
+    run_job,
+    setup_globals,
+)
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.search_path_plugin import SearchPathPlugin
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, open_dict
+import subprocess
+import os
+import sys
+import hydra
+import time
+import copy
+
+from joblib import Parallel, delayed
+
+# IMPORTANT:
+# If your plugin imports any module that takes more than a fraction of a second to import,
+# Import the module lazily (typically inside launch()).
+# Installed plugins are imported during Hydra initialization and plugins that are slow to import plugins will slow
+# the startup of ALL hydra applications.
+# Another approach is to place heavy includes in a file prefixed by _, such as _core.py:
+# Hydra will not look for plugin in such files and will not import them during plugin discovery.
+
+
+log = logging.getLogger(__name__)
+
+def look_for_task_spooler():
+    """
+    Task spooler is known as `tsp` on Ubuntu and `ts` on MacOS. 
+    This function checks for which one exists and returns the
+    valid command, or throws an error if neither is found.
+
+    Returns:
+        str: Either `ts` or `tsp` depending on which is valid.
+
+    Raises:
+        RuntimeError when TaskSpooler isn't found on the system.
+    """
+    commands = {'ts': False, 'tsp': False}
+
+    for c in commands:
+        try:
+            cmd_output = str(subprocess.check_output([c, '-h']))
+            if 'usage' in cmd_output:
+                commands[c] = True
+        except:
+            commands[c] = False
+    
+    if not any(commands.values()):
+        raise RuntimeError("Task spooler not found! Install it to use this launcher.")
+    
+    for c in commands:
+        if commands[c]:
+            return c
+
+def _get_job_status(tsp_prefix, job_id):
+    job_status = subprocess.check_output(
+        f"{tsp_prefix} -s {job_id}", shell=True)
+    return job_status.rstrip().decode('utf-8')
+
+def execute_job(
```

Comment:
this function will need some work.
we should do something similar to what submitit is doing here.